pkg/webui/components/overlay/index.js

```diff
@@ -22,15 +22,15 @@ import Message from '../../lib/components/message'
 
 import style from './overlay.styl'
 
-const Overlay = ({ className, visible, loading, children }) => (
+const Overlay = ({ className, overlayClassName, spinnerClassName, visible, loading, children }) => (
   <div className={classnames(className, style.overlayWrapper)}>
     <div
-      className={classnames(style.overlay, {
+      className={classnames(overlayClassName, style.overlay, {
         [style.overlayVisible]: visible,
       })}
     />
     {visible && loading && (
-      <Spinner center>
+      <Spinner center className={spinnerClassName}>
         <Message content={sharedMessages.fetching} />
```

Comment:
The message of the spinner should be passable (while defaulting to `sharedMessages.fetching`). In case of the location map it should say `Loading location…`